File: app-java-fx/src/main/java/chessgame/facade/mock/MockBoardFacade.java
package chessgame.facade.mock;

import chessgame.facade.BoardFacade;
import chessgame.facade.PieceFacade;

import java.util.LinkedHashMap;
import java.util.Map;

public class MockBoardFacade implements BoardFacade {

    @Override
    public Map<String, PieceFacade> initBlackPieces() {
        final Map<String, PieceFacade> map = new LinkedHashMap<>();

        map.put("A8", () -> "\u265C");
        map.put("B8", () -> "\u265E");
        map.put("C8", () -> "\u265D");
        map.put("D8", () -> "\u265B");
        map.put("E8", () -> "\u265A");
        map.put("F8", () -> "\u265D");
        map.put("G8", () -> "\u265E");
        map.put("H8", () -> "\u265C");
        map.put("A7", () -> "\u265F");
        map.put("B7", () -> "\u265F");
        map.put("C7", () -> "\u265F");
        map.put("D7", () -> "\u265F");
        map.put("E7", () -> "\u265F");
        map.put("F7", () -> "\u265F");
        map.put("G7", () -> "\u265F");
        map.put("H7", () -> "\u265F");

        return map;
    }

    @Override
    public Map<String, PieceFacade> initWhitePieces() {
        final Map<String, PieceFacade> map = new LinkedHashMap<>();

        map.put("A1", () -> "\u2656");
        map.put("B1", () -> "\u2658");
        map.put("C1", () -> "\u2657");
        map.put("D1", () -> "\u2655");
        map.put("E1", () -> "\u2654");
        map.put("F1", () -> "\u2657");
        map.put("G1", () -> "\u2658");
        map.put("H1", () -> "\u2656");
        map.put("A2", () -> "\u2659");
        map.put("B2", () -> "\u2659");
        map.put("C2", () -> "\u2659");
        map.put("D2", () -> "\u2659");
        map.put("E2", () -> "\u2659");
        map.put("F2", () -> "\u2659");
        map.put("G2", () -> "\u2659");
        map.put("H2", () -> "\u2659");

        return map;
    }
}

File: app-kotlin-fx/src/main/kotlin/chessgame/facade/mock/MockBoardFacade.kt
package chessgame.facade.mock

import chessgame.facade.BoardFacade

class MockBoardFacade: BoardFacade {

    override fun initBlackPieces(): Map<String, String> = mapOf(
            "A8" to "\u265C",
            "B8" to "\u265E",
            "C8" to "\u265D",
            "D8" to "\u265B",
            "E8" to "\u265A",
            "F8" to "\u265D",
            "G8" to "\u265E",
            "H8" to "\u265C",
            "A7" to "\u265F",
            "B7" to "\u265F",
            "C7" to "\u265F",
            "D7" to "\u265F",
            "E7" to "\u265F",
            "F7" to "\u265F",
            "G7" to "\u265F",
            "H7" to "\u265F")

    override fun initWhitePieces(): Map<String, String> = mapOf(
            "A1" to "\u2656",
            "B1" to "\u2658",
            "C1" to "\u2657",
            "D1" to "\u2655",
            "E1" to "\u2654",
            "F1" to "\u2657",
            "G1" to "\u2658",
            "H1" to "\u2656",
            "A2" to "\u2659",
            "B2" to "\u2659",
            "C2" to "\u2659",
            "D2" to "\u2659",
            "E2" to "\u2659",
            "F2" to "\u2659",
            "G2" to "\u2659",
            "H2" to "\u2659")
}
